src/js/components/List/index.d.ts

```diff
@@ -54,9 +58,9 @@ export interface ListProps<ListItemType> {
   };
   margin?: MarginType;
   onActive?: (index: number) => void;
-  onClickItem?:
-    | ((event: React.MouseEvent) => void)
-    | ((event: { item?: ListItemType; index?: number }) => void);
+  onClickItem?: (
+    event: React.MouseEvent & { item: ListItemType; index: number },
```

Comment:
should this be optional? 
`event: React.MouseEvent & { item?: ListItemType; index?: number },`

Reply:
Not according to [the documentation](https://v2.grommet.io/list#onClickItem), but happy to change if desired

Reply:
I think that's fine I read through and matched the documentation.